share/p2p/discovery/discovery.go

```diff
@@ -114,7 +114,7 @@ func (d *Discovery) Start(context.Context) error {
 
 func (d *Discovery) Stop(context.Context) error {
 	d.cancel()
-	return nil
+	return d.metrics.close()
```

Comment:
Here error from from closing metrics is propagated from `Stop()`.But in In some modules error is only logged and not returned. Example:
```
if err := d.sampler.metrics.close(); err != nil {
	log.Errorw("closing metrics", "err", err)
}
```
Also in cases when metrics unregistered directly by fx.LifeCycle it is also propagated to fx. 
Lets make it consistent and not propagate, as it does not matter if unregister failed. Also log severity cpuld be lowered to WARN from ERROR for the same reason.

Reply:
https://github.com/celestiaorg/celestia-node/pull/3281/commits/3ffcdeb2644a3ef07a891c59a57dc7d87fdfd3e1